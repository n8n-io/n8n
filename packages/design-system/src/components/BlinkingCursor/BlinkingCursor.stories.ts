import BlinkingCursor from './BlinkingCursor.vue';
import type { StoryFn } from '@storybook/vue3';

export default {
	title: 'Assistant/BlinkingCursor',
	component: BlinkingCursor,
	argTypes: {},
};

const Template: StoryFn = (args, { argTypes }) => ({
	setup: () => ({ args }),
	props: Object.keys(argTypes),
	components: {
		BlinkingCursor,
	},
	template: '<blinking-cursor v-bind="args" />',
});

export const Cursor = Template.bind({});
