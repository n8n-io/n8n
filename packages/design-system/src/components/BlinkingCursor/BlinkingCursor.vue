<template>
	<span class="blinking-cursor"></span>
</template>

<style lang="scss">
.blinking-cursor {
	display: inline-block;
	height: var(--font-size-m);
	width: var(--spacing-3xs);
	border-radius: var(--border-radius-small);
	margin-left: var(--spacing-4xs);

	animation: 1s blink step-end infinite;
}

@keyframes blink {
	from,
	to {
		background-color: transparent;
	}
	50% {
		background-color: var(--color-foreground-xdark);
	}
}
</style>
