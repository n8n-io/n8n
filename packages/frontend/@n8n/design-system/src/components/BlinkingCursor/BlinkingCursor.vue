<template>
	<span class="blinking-cursor"></span>
</template>

<style lang="scss">
.blinking-cursor {
	display: inline-block;
	height: var(--font-size--md);
	width: var(--spacing--3xs);
	border-radius: var(--radius--sm);
	margin-left: var(--spacing--4xs);

	animation: 1s blink step-end infinite;
}

@keyframes blink {
	from,
	to {
		background-color: transparent;
	}
	50% {
		background-color: var(--color--foreground--shade-2);
	}
}
</style>
