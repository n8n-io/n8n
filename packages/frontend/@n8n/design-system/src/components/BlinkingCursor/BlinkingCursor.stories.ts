import type { StoryFn } from '@storybook/vue3';

import BlinkingCursor from './BlinkingCursor.vue';

export default {
	title: 'Assistant/BlinkingCursor',
	component: BlinkingCursor,
	argTypes: {},
};

const Template: StoryFn = (args, { argTypes }) => ({
	setup: () => ({ args }),
	props: Object.keys(argTypes),
	components: {
		BlinkingCursor,
	},
	template: '<blinking-cursor v-bind="args" />',
});

export const Cursor = Template.bind({});
