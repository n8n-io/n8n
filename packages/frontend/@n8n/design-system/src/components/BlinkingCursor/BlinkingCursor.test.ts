import { render } from '@testing-library/vue';

import BlinkingCursor from './BlinkingCursor.vue';

describe('BlinkingCursor', () => {
	it('renders blinking cursor correctly', () => {
		const { container } = render(BlinkingCursor, {});
		expect(container).toMatchSnapshot();
	});
});
