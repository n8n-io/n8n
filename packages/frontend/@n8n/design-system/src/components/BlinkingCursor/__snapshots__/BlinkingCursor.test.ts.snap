// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlinkingCursor > renders blinking cursor correctly 1`] = `
<div>
  <span
    class="blinking-cursor"
  />
</div>
`;
